main/src/cgeo/geocaching/models/Geocache.java

```diff
@@ -1652,32 +1652,29 @@ public boolean addWaypointsFromText(@Nullable final String text, final boolean u
         return changed;
     }
 
-    private Waypoint findWaypoint(final Waypoint searchWp) {
-        //try to match prefix
+    @Nullable
+    private Waypoint findWaypoint(@NonNull final Waypoint searchWp) {
+        // try to match prefix
         final String prefix = searchWp.getPrefix();
-        if (null != prefix) {
-            for (final Waypoint waypoint : waypoints) {
-                if (!StringUtils.isBlank(prefix) && !StringUtils.isBlank(waypoint.getPrefix()) && prefix.equals(waypoint.getPrefix())) {
-                    return waypoint;
-                }
+        for (final Waypoint waypoint : waypoints) {
```

Comment:
If you put 
 `if (!StringUtils.isBlank(prefix)) ` 
before the for-loop, you can avoid the then useless loop.